PR for Documentation change proposals

stisiTT opened the pull request:
PR for Documentation suggestions from this issue: https://github.com/tenstorrent/tt-metal/issues/17394

stisiTT committed:
Small QOL suggestions to docs
Changing comments on shell examples for clarity

stisiTT committed:
Move repo-clone to Step 2
Can't use install_dependencies.sh without repo, and it's required regardless of install option, so should be earlier than where it's currently stated.

tt-rkim commented on INSTALLING.md:
```diff
@@ -64,12 +66,12 @@ Set up the environment variables and invoke our build scripts. Note that for
 ./build_metal.sh
 
-# If you would like an out-of-the-box virtual environment to use,
+# See NOTE below on Python Environments for alternative to OOTB venv 
```
What's OOTB?

stisiTT replied:
Out of the box. I was attempting to use the same language that is in the note being referred to here so there was consistency there, but I also was being mindful of having too many characters on the line

tt-rkim reviewed (commented):
Minor nitpick comment asking for explanation

cc: @bbeggsTT @gsarabandoTT  - please a take look

stisiTT committed:
Moving Validation step outside of Docker-specific subheader and re-numbering steps to stop skipping 4.

Validation could occur regardless of if Option 1, 2, or 3 are used.

Validation also previously labelled as a 2nd step in Option 3 where the 1st. step was in Option 2, seemingly having one process branching over two disctint build options. Option 2 had its '1.' changed to a smaller sub header and '2.' inside of Option 3 was changed into a bigger subheader to promote validation across all builds.

Step 5 (Optional) exists where there is no Step 4. Step 5 renamed to Step 4.

stisiTT committed:
Setting venv specifically for Wormhole

The WH_ARCH_YAML is a variable used by tt-metal to instruct the runtime to dispatch specific kernels to specific tensix cores on the core grid. It is required for n300 and not n150 because n300 cards have different "harvesting" specs. Specifically they harvest two rows of the tensix grid instead of 1 row like the n150 does.
Setting this variable to that value informs tt-metal of the location of the specific ETH cores to use for dispatch.

gsarabandoTT commented on INSTALLING.md:
````diff
@@ -64,19 +66,29 @@ Set up the environment variables and invoke our build scripts. Note that for
+```sh
+# See NOTE below on Python Environments for alternative to OOTB venv
 ./create_venv.sh
 source python_env/bin/activate
 ```
 
-where `ARCH_NAME` is one of `grayskull`, `wormhole_b0`, or `blackhole`,
-depending on your Tenstorrent card type.
-
 > [!NOTE]
 >
 > Note about Python environments: You do not have to use `create_venv.sh`. If you
````
We've been working on a similar PR for INSTALLING.md. This note on python environments can be removed, just keep one instruction such as: 
"(recommended) For an out-of-the-box virtual environment, execute:". Folks agreed that anything beyond what we recomend is out of our scope.

gsarabandoTT replied:
Here is the similar PR for reference: [tt-metal/pull/15800](https://github.com/tenstorrent/tt-metal/pull/15800)

tt-rkim commented:
Any updates on this?

tt-rkim commented:
Perhaps another line saying that if you want to work with the source in a docker container, rather than just the API, then you should also clone the source within the container.

You can close this. You can make a separate one to reconcile this source-in-docker thing you're concerned about.
